### ISSUE TITLE
Add an option to ignore incompatible kernel

### ISSUE THREAD

**Issue by archandanime** [chenxiaolong/avbroot#108]

**Comment by archandanime** [comment]
I tried to patch Pixel 4 XL(coral) with KernelSU patched boot image [pixel4xl_android13_4.14.276_v061.img](https://github.com/msnx/KernelSU-Pixel4XL/releases/tag/v0.6.1) in my script using:
```
python avbroot/avbroot.py patch \
--input ${OTA_ZIP} \
--privkey-avb ${AVB_KEYFILE} \
--privkey-ota ${OTA_KEYFILE} \
--cert-ota ${OTA_CRT} \
--prepatched ${KERNELSU_BOOT_IMG} \
--boot-partition @gki_kernel \
--clear-vbmeta-flags \
--passphrase-avb-env-var PASSPHRASE_AVB \
--passphrase-ota-env-var PASSPHRASE_OTA \
--output ${PATCHED_OTA_ZIP}
```

but it ended up with:

```
    raise ValueError('The prepatched boot image is not compatible '
ValueError: The prepatched boot image is not compatible with the original:
- id header field was changed: b"{6\xe3UW\xdd\x17exW\x94\xd9k\x95\xd8\xe5'\x15N\xb8\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" -> b'\xc8\x95\xb2l\xde~\xcc\xa6,o\xd4L\x81\x0e\xd3r\x1c\xa4\xdb.\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
- cmdline header field was changed: b'console=ttyMSM0,115200n8 androidboot.console=ttyMSM0 printk.devkmsg=on msm_rtb.filter=0x237 ehci-hcd.park=3 service_locator.enable=1 androidboot.memcg=1 cgroup.memory=nokmem usbcore.autosuspend=7 androidboot.usbcontroller=a600000.dwc3 swiotlb=2048 androidboot.boot_devices=soc/1d84000.ufshc loop.max_part=7 cgroup_disable=pressure buildvariant=userdebug' -> b'console=ttyMSM0,115200n8 androidboot.console=ttyMSM0 printk.devkmsg=on msm_rtb.filter=0x237 ehci-hcd.park=3 service_locator.enable=1 androidboot.memcg=1 cgroup.memory=nokmem usbcore.autosuspend=7 androidboot.usbcontroller=a600000.dwc3 swiotlb=2048 androidboot.boot_devices=soc/1d84000.ufshc loop.max_part=7 buildvariant=user'
- os_version header field was changed: 436207990 -> 0
```
a while ago you made commit [#67](https://github.com/chenxiaolong/avbroot/pull/67)

**Comment by archandanime** [comment]
I apology for multiple edits and closing/re-opening the issues, I... accidentally hit Enter key multiple times.

**Comment by chenxiaolong** [comment]
Ah, this is a good suggestion. I think I'll make a couple changes:

* The default behavior will have slightly relaxed checks. The `id` and `os_version` fields you mentioned are meaningless to the bootloader, so there's no reason to check them. `cmdline` will still be checked by default since it's easy to make the device unbootable with a bad kernel command line.
* For ignoring warnings, I'll add something like `--ignore-prepatched-compat`. If it's specified once, it'll ignore a "safer" subset of checks (like `cmdline`). If it's specified twice, it'll ignore all checks.

  It's a bit annoying to potentially need to specify `--ignore-prepatched-compat --ignore-prepatched-compat`, but I want to make sure it's hard to use a boot image that is unlikely to work (eg. incompatible KMI version). Since there's at least one ROM that auto-disables the OEM unlocking switch, avoiding unbootable systems is a top priority for the project.

**Comment by chenxiaolong** [comment]
This has been implemented in #112.
